pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -282,7 +282,7 @@ public boolean readCompacted() {
 
         // calculate avg message per entry
         int tmpAvgMessagesPerEntry = AVG_MESSAGES_PER_ENTRY.get(this);
-        tmpAvgMessagesPerEntry = (int) Math.round(tmpAvgMessagesPerEntry * avgPercent
+        tmpAvgMessagesPerEntry = (int) Math.floor(tmpAvgMessagesPerEntry * avgPercent
```

Comment:
I don't think using `floor` would solve this kind of precision issue.
Using `floor` also have problems of this value going up, e.g. `avgMessagesPerEntry` = 1 and new value is 2. The result is always 1.

We should just use double type for `avgMessagesPerEntry`.

Reply:
+1

Reply:
no, we should not use double which will not solve it. 
floor solves it. mathematically `round` was calculating higher value for `5.5 to -> 6` and formula `(tmpAvgMessagesPerEntry * avgPercent + (1 - avgPercent) * totalMessages / entries.size())` always results into `6 -> 5.5`. therefore, result always stays the same`6`.
`floor` returns the largest less than or equal to the argument which makes value keep going down and eventually becomes 1. It fixes the issue into the right way.
Please test and unblock the PR.

Reply:
> no, we should not use double which will not solve it.

Why not? Please take another look.

This fix solve the case from 6 to 1, but aggravated the case like that the value should increase from 1 to 6.